manifest/manifest.go

```diff
@@ -39,18 +39,15 @@ var (
 	}
 )
 
-type OnManifestChange func(ctx context.Context, prevManifest Manifest, errCh chan error)
+type OnManifestChange func(ctx context.Context, prevManifest Manifest) error
 
 type ManifestProvider interface {
-	// Run starts any background tasks required for the operation
-	// of the manifest provider.
-	Run(context.Context, chan error)
```

Comment:
Renamed to start because it doesn't block.

Reply:
I also removed the error channels here and everywhere else because there was no way to use them reliably. Specifically, there was no way to know if we had received all the errors so we couldn't receive an error then exit. And if we ever stopped reading on the channel, various components would block.

I guess one solution would have been to write to this error channel with a context... but I just got rid of it and switched to error groups instead.